src/main/java/org/folio/rest/impl/PermsAPI.java

```diff
@@ -1937,4 +1937,28 @@ private static Criterion buildPermissionNameListQuery(List<String> permissionNam
     }
     return criterion;
   }
+
+  @Override
+  public void postPermsPurgeDeprecated(Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+      Context vertxContext) {
+    try {
+      String tenantId = TenantTool.tenantId(okapiHeaders);
+      PostgresClient pgClient = PostgresClient.getInstance(vertxContext.owner(), tenantId);
+      PermissionUtils.purgeDeprecatedPermissions(pgClient, tenantId)
+        .onSuccess(res -> {
+          asyncResultHandler
+            .handle(Future.succeededFuture(PostPermsPurgeDeprecatedResponse.respond200WithApplicationJson(res)));
+        })
+        .onFailure(ex -> {
+          asyncResultHandler
+            .handle(Future.succeededFuture(PostPermsPurgeDeprecatedResponse.respond500WithTextPlain(ex.getCause()
+              .getMessage())));
```

Comment:
`ex.getCause()` may be null. Use `ex.getMessage()`
```suggestion
          asyncResultHandler
            .handle(Future.succeededFuture(PostPermsPurgeDeprecatedResponse.respond500WithTextPlain(ex.getMessage())));
```

Reply:
Make sense. Changed.